lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -235,4 +235,35 @@ sub save_results {
     path($self->job->result_dir, 'details-' . $self->name . '.json')->spurt(encode_json($results));
 }
 
+# Collect text output into JSON. save_results() is called too early for this.
+sub finalize_results {
+    my ($self) = @_;
+
+    my $dir = $self->job->result_dir();
+    return unless $dir;
+    my $file = path($dir, "details-" . $self->name . ".json");
+    return unless -e $file;
+    my $results;
+
+    try {
+        $results = decode_json($file->slurp);
+    }
+    catch {
+        return;
+    };
+
+    $results = {details => $results} if ref($results) eq 'ARRAY';
+
+    for my $step (@{$results->{details}}) {
+        next unless $step->{text};
+        my $txtfile = path($dir, $step->{text});
+        next unless -e $txtfile;
+        $step->{text_data} = $txtfile->slurp;
```

Comment:
Of course it would be nice to get rid of these text files in the end and not leave the data duplicated. Since the text files are still used within the "step view" that's not so easy of course.